### PR TITLE
build libspatialindex for experimental platforms, bump to 1.9.3

### PR DESCRIPTION
**Pull request by mattwigway**
This updates the LibSpatialIndex_jll package to build for experimental platforms, and additionally updates to version 1.9.3. It should fix JuliaGeo/LibSpatialIndex.jl#15

### CODE REVIEW

**Comment by mattwigway** [comment]
Thanks for the suggestions @giordano, will integrate those!

I'm not sure why the Windows builds are failing. On BuildKite it's not showing any errors other than that the libraries are missing, but it also shows just above that that the libraries were found, so I guess something to do with the auditor? When I run locally with `--verbose` I get `Warning: Minimum instruction set detected for libspatialindex.dll is avx, not x86_64 as desired.` which is probably related. I'm not sure how to fix that - shouldn't the CMake toolchain file specify the machine architecture?

**Comment by imciner2** [comment]
> I'm not sure why the Windows builds are failing. On BuildKite it's not showing any errors other than that the libraries are missing

The installation log seems to suggest they are being installed wrong. Here is where they are installed:

```
[17:21:56] -- Installing: /workspace/destdir/lib/libspatialindex.dll.a
[17:21:56] -- Installing: /workspace/destdir/./libspatialindex.dll
[17:21:56] -- Installing: /workspace/destdir/lib/libspatialindex_c.dll.a
[17:21:56] -- Installing: /workspace/destdir/./libspatialindex_c.dll
```
But they should instead be inside `${destdir}/bin`, so something in the toolchain has converted the `bin` to `.`

**Comment by giordano** [comment]
Don't ask me why: https://github.com/libspatialindex/libspatialindex/blob/3be67583852c741d0357f7ef5148c5fe8b484076/CMakeLists.txt#L181

**Comment by mattwigway** [comment]
I think this is good to go on the technical side, and all the LibSpatialIndex.jl tests pass on my M1, but it would be great if one of the LibSpatialIndex.jl maintainers could comment before merge (cc @yeesian @visr @evetion) since there's a version bump as well as the compile change, and it increases the compat entry for Julia to 1.6. Based on the LibSpatialIndex changelog, it looks like the only changes from 1.8.5 are bugfixes and new features, so I don't think anything should stop working.

**Review by visr — APPROVED** [review]
Thanks for taking this on!

**Comment by evetion** [comment]
Thanks from a happy M1 user! Libspatialindex already requires Julia 1.6, so this should be no problem.

File: L/LibSpatialIndex/build_tarballs.jl
Comment (giordano) [inline L/LibSpatialIndex/build_tarballs.jl:38]:
```suggestion
platforms = expand_cxxstring_abis(supported_platforms())
```

File: L/LibSpatialIndex/build_tarballs.jl
Comment (giordano) [inline L/LibSpatialIndex/build_tarballs.jl:31]:
```suggestion
cmake --build . -j${nproc}
```

**Comment by giordano** [comment]
Thanks for your contribution!

**Pull request merged**
